### PR TITLE
Enhancement/25 unit tests

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**

### CODE REVIEW

File: src/test/java/com/exasol/adapter/sql/SqlLiteralIntervalTest.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    @Test
+    void testGetValue() {
+        assertThat(sqlLiteralIntervalDayToSecond.getValue(), equalTo(VALUE));
+    }
+
+    @Test
+    void testToSimpleSql() {
+        assertThat(sqlLiteralIntervalDayToSecond.toSimpleSql(), equalTo("INTERVAL '" + VALUE + "' DAY (1) TO SECOND (2)"));
```
Comment (redcatbear) [inline src/test/java/com/exasol/adapter/sql/SqlLiteralIntervalTest.java]: Please split into two tests.
Please also test with fractions of seconds. Parameterized JUnit 5 tests can make this more comfortable and extensible.

File: src/test/java/com/exasol/adapter/sql/SqlPredicateAndTest.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        andedPredicates = new ArrayList<>();
+        andedPredicates.add(new SqlLiteralNull());
+        andedPredicates.add(new SqlLiteralBool(true));
+        sqlPredicateAnd = new SqlPredicateAnd(andedPredicates);
+        sqlPredicateAndWithEmptyList = new SqlPredicateAnd(Collections.emptyList());
+    }
+
+    @Test
+    void testGetAndedPredicates() {
+        assertThat(sqlPredicateAndWithEmptyList.getAndedPredicates(), equalTo(Collections.emptyList()));
```
Comment (redcatbear) [inline src/test/java/com/exasol/adapter/sql/SqlPredicateAndTest.java]: Split into two tests please.

File: src/test/java/com/exasol/adapter/sql/SqlUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    @Test
+    void testQuoteIdentifierIfNeededWithoutQuotes() {
+        assertThat(SqlUtils.quoteIdentifierIfNeeded("TABLE", Collections.emptyMap()), equalTo(TABLE_WITHOUT_QUOTES));
+    }
+
+    @Test
+    void testQuoteIdentifierIfNeededWithQuotes() {
+        final Map<String, String> map = new HashMap<>();
+        map.put("QUOTE_CHAR", "_");
+        assertThat(SqlUtils.quoteIdentifierIfNeeded("TaBlE", map), equalTo(TABLE_WITH_QUOTES));
```
Comment (redcatbear) [inline src/test/java/com/exasol/adapter/sql/SqlUtilsTest.java]: Split into two tests please.

File: src/test/java/com/exasol/utils/JsonHelperTest.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Test
+    void testGetJsonObject() {
+        assertThat(JsonHelper.getJsonObject(JSON), equalTo(expectedJsonObject));
+    }
+
+    @Test
+    void testGetKeyAsString() {
+        assertThat(JsonHelper.getKeyAsString(jsonObject, KEY_1, DEFAULT_VALUE), equalTo(DEFAULT_VALUE));
```
Comment (redcatbear) [inline src/test/java/com/exasol/utils/JsonHelperTest.java]: Split into two tests.

File: pom.xml [hunk 5]
```diff
@@ -64,14 +64,44 @@
...
+            <version>5.3.2</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.platform</groupId>
+            <artifactId>junit-platform-runner</artifactId>
+            <version>1.3.2</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply pom.xml]: The ticket is already created

File: src/test/java/com/exasol/adapter/json/ResponseJsonSerializerTest.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Test
+    void testMakeGetCapabilitiesResponse() throws JSONException {
+        final Capabilities capabilities = new Capabilities();
+        capabilities.supportMainCapability(MainCapability.LIMIT);
+        capabilities.supportLiteral(LiteralCapability.DATE);
+        capabilities.supportPredicate(PredicateCapability.EQUAL);
+        capabilities.supportScalarFunction(ScalarFunctionCapability.ADD);
+        capabilities.supportAggregateFunction(AggregateFunctionCapability.AVG);
+        JSONAssert.assertEquals("{\"type\":\"getCapabilities\"," + "\"capabilities\":[\"LIMIT\"," + "\"LITERAL_DATE\"," + "\"FN_PRED_EQUAL\", " + "\"FN_AGG_AVG\", " + "\"FN_ADD\"]}", ResponseJsonSerializer.makeGetCapabilitiesResponse(capabilities), false);
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/json/ResponseJsonSerializerTest.java]: done

File: src/test/java/com/exasol/adapter/json/SqlDataTypeJsonSerializerTest.java [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+        when(dataType.getScale()).thenReturn(3);
+        final JsonObject jsonObject = SqlDataTypeJsonSerializer.serialize(dataType).build();
+        assertThat(jsonObject.getInt("precision"), equalTo(5));
+        assertThat(jsonObject.getInt("scale"), equalTo(3));
+    }
+
+    @Test
+    void testSerializeDoubleDateAndBoolean() {
+        when(dataType.getExaDataType()).thenReturn(DataType.ExaDataType.DOUBLE);
+        assertAll(() -> SqlDataTypeJsonSerializer.serialize(dataType));
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/json/SqlDataTypeJsonSerializerTest.java]: done

File: src/test/java/com/exasol/adapter/json/SqlDataTypeJsonSerializerTest.java [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        when(dataType.getExaDataType()).thenReturn(DataType.ExaDataType.DECIMAL);
+        when(dataType.getPrecision()).thenReturn(5);
+        when(dataType.getScale()).thenReturn(3);
+        final JsonObject jsonObject = SqlDataTypeJsonSerializer.serialize(dataType).build();
+        assertThat(jsonObject.getInt("precision"), equalTo(5));
+        assertThat(jsonObject.getInt("scale"), equalTo(3));
+    }
+
+    @Test
+    void testSerializeDoubleDateAndBoolean() {
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/json/SqlDataTypeJsonSerializerTest.java]: done

File: src/test/java/com/exasol/adapter/metadata/ColumnMetadataTest.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    private static final String TEST_ADAPTER_NOTES = "testAdapterNotes";
+    private static final boolean NULLABLE = true;
+    private static final boolean IS_IDENTITY = true;
+    private static final String TEST_DEFAULT_VALUE = "testDefaultValue";
+    private static final String TEST_COMMENT = "testComment";
+    private ColumnMetadata columnMetadata;
+    private final DataType dataType = DataType.createDouble();
+
+    @BeforeEach
+    void SetUp() {
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/metadata/ColumnMetadataTest.java]: done

File: src/test/java/com/exasol/adapter/metadata/ColumnMetadataTest.java [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+    @Test
+    void testGetComment() {
+        assertThat(columnMetadata.getComment(), equalTo(TEST_COMMENT));
+    }
+
+    @Test
+    void testHasNoDefaultValueAndComment() {
+        final ColumnMetadata columnMetadata2 = new ColumnMetadata(TEST_NAME, TEST_ADAPTER_NOTES, dataType, NULLABLE, IS_IDENTITY, "", "");
+        assertThat(columnMetadata2.hasDefault(), equalTo(false));
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/metadata/ColumnMetadataTest.java]: done

File: src/test/java/com/exasol/adapter/request/RefreshRequestTest.java [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+@MockitoSettings(strictness = Strictness.LENIENT)
+class RefreshRequestTest {
+    @Mock
+    private SchemaMetadataInfo schemaMetadataInfo;
+    private RefreshRequest refreshRequest;
+    private List<String> tables;
+
+    @BeforeEach
+    void SetUp() throws AdapterException {
+        tables = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/request/RefreshRequestTest.java]: done

File: src/test/java/com/exasol/adapter/request/RefreshRequestTest.java [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+@RunWith(JUnitPlatform.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class RefreshRequestTest {
+    @Mock
+    private SchemaMetadataInfo schemaMetadataInfo;
+    private RefreshRequest refreshRequest;
+    private List<String> tables;
+
+    @BeforeEach
+    void SetUp() throws AdapterException {
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/request/RefreshRequestTest.java]: done

File: src/test/java/com/exasol/adapter/sql/SqlArgumentValidatorTest.java [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class SqlArgumentValidatorTest {
+    @Test
+    void testValidateSqlFunctionArgumentsNull() {
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/sql/SqlArgumentValidatorTest.java]: done

File: src/test/java/com/exasol/adapter/sql/SqlFunctionScalarCaseTest.java [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+        arguments.add(new SqlLiteralNull());
+        results = new ArrayList<>();
+        results.add(new SqlLiteralNull());
+        basis = new SqlLiteralNull();
+        sqlFunctionScalarCase = new SqlFunctionScalarCase(arguments, results, basis);
+    }
+
+    @Test
+    void testGetArguments() {
+        assertThat(sqlFunctionScalarCase.getArguments(), equalTo(arguments));
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/sql/SqlFunctionScalarCaseTest.java]: done

File: src/test/java/com/exasol/adapter/sql/SqlPredicateAndTest.java (hunk 2 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/sql/SqlPredicateAndTest.java]: done

File: src/test/java/com/exasol/utils/JsonHelperTest.java (hunk 4 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/utils/JsonHelperTest.java]: done

File: src/test/java/com/exasol/adapter/sql/SqlUtilsTest.java (hunk 3 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/sql/SqlUtilsTest.java]: done

File: src/test/java/com/exasol/adapter/sql/SqlLiteralIntervalTest.java (hunk 1 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/sql/SqlLiteralIntervalTest.java]: done

File: src/test/java/com/exasol/adapter/json/SqlDataTypeJsonSerializerTest.java [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+              () -> SqlDataTypeJsonSerializer.serialize(this.dataType));
+    }
+
+    @Test
+    void testSerializeDecimal() {
+        when(this.dataType.getExaDataType()).thenReturn(DataType.ExaDataType.DECIMAL);
+        when(this.dataType.getPrecision()).thenReturn(5);
+        when(this.dataType.getScale()).thenReturn(3);
+        final JsonObject jsonObject = SqlDataTypeJsonSerializer.serialize(this.dataType).build();
+        assertThat(jsonObject.getInt("precision"), equalTo(5));
```
Comment (redcatbear) [inline src/test/java/com/exasol/adapter/json/SqlDataTypeJsonSerializerTest.java]: Please use `assertAll` around the two sub tests.

Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/json/SqlDataTypeJsonSerializerTest.java]: done

**Pull request merged**